zk-collectd.py

```diff
@@ -43,30 +44,55 @@ def __init__(self, host='localhost', port='2181', timeout=1):
 
     def get_stats(self):
         """Get ZooKeeper server stats as a map."""
-        data = self._send_cmd('mntr')
-        return self._parse(data)
+        stats = {}
+        for cmd in ZK_CMDS:
+            data = self._send_cmd(cmd)
+            if cmd == "ruok":
+                results = self._parse_ok(data)
```

Comment:
So, again, if you have to special case here inside a loop, it makes using a loop somewhat meaningless. Just do:

```python
def get_stats(self):
    stats = {}
    stats.update(self._get_health_stats())
    stats.update(self._get_mntr_stats())
    # We can add more metrics here in the future.
    return stats

def _get_health_stat(self):
    response = self._send_cmd(RUOK_CMD)
    return {'zk_service_health': int(response == IMOK_RESP)}

def _get_mntr_stats(self):
    response = self._send_cmd(MNTR_CMD)
    result = {}
    for line in response.splitlines():
        try:
            key, value = self._parse_line(line)
            if key == 'zk_server_state':
                result['zk_is_leader'] = int(value != 'follower')
            elif key == 'zk_version':
                continue
            else:
                result[key] = value
    return result
```